src/dev/failed_tests/report.js

```diff
@@ -0,0 +1,149 @@
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import xml2js from 'xml2js';
+import vfs from 'vinyl-fs';
+import es from 'event-stream';
+import { getGithubClient, markdownMetadata, paginate } from '../github_utils';
+import { find } from 'lodash';
+import stripAnsi from 'strip-ansi';
+
+const GITHUB_FLAKY_TEST_LABEL = 'flaky-failing-test';
+const GITHUB_OWNER = 'elastic';
+const GITHUB_REPO = 'kibana';
+const BUILD_URL = process.env.BUILD_URL;
+
+/**
+ * Parses junit XML files into JSON
+ */
+const mapXml = es.map((file, cb) => {
+  xml2js.parseString(file.contents.toString(), (err, result) => {
+    cb(null, result);
+  });
+});
+
+/**
+ * Filters all testsuites to find failed testcases
+ */
+const filterFailures = es.map((testSuite, cb) => {
+  const testFiles = testSuite.testsuites.testsuite;
+
+  const failures = testFiles.reduce((failures, testFile) => {
+    for (const testCase of testFile.testcase) {
+      if (testCase.failure) {
+        // unwrap xml weirdness
+        failures.push({
+          ...testCase.$,
+          // Strip ANSI color characters
+          failure: stripAnsi(testCase.failure[0])
+        });
+      }
+    }
+
+    return failures;
+  }, []);
+
+  console.log(`Found ${failures.length} test failures`);
+
+  cb(null, failures);
+});
+
+/**
+ * Creates and updates github issues for the given testcase failures.
+ */
+const updateGithubIssues = (githubClient, issues) => {
+  return es.map(async (failureCases, cb) => {
+
+    const issueOps = failureCases.map(async (failureCase) => {
+      const existingIssue = find(issues, (issue) => {
+        return markdownMetadata.get(issue.body, 'test.class') === failureCase.classname &&
+          markdownMetadata.get(issue.body, 'test.name') === failureCase.name;
+      });
+
+      if (existingIssue) {
+        // Increment failCount
+        const newCount = (markdownMetadata.get(existingIssue.body, 'test.failCount') || 0) + 1;
+        const newBody = markdownMetadata.set(existingIssue.body, 'test.failCount', newCount);
+
+        await githubClient.issues.edit({
+          owner: GITHUB_OWNER,
+          repo: GITHUB_REPO,
+          number: existingIssue.number,
+          state: 'open',  // Reopen issue if it was closed.
```

Comment:
I am leaning towards creating a new issue here. The reason for the failure could be different than previous failures. Thoughts?

Reply:
It could be, but I think it's also more likely that we _think_ we fixed a test and closed the issue just for it to fail a few days later (maybe it only fails 1% of the time). In that case, it seems useful to know that it failed recently and to be able to quickly see what we tried to fix it.

I can't think of a great reason that the historical context for a given test failure would be bad to surface up.

Reply:
Ok - we can stick with this and always change later. It's possible we could also create an arbitrary timeout. So after an issue was closed for X amount of time a new issue is created.